src/odemis/gui/cont/tabs.py

```diff
@@ -4389,7 +4393,481 @@ def _on_align_move_done(self, f=None):
+        :param proportion (int): direction of the movement represented by the button (-1/1)
+        :param evt (GenButtonEvent): Clicked event
+        """
+        # Only proceed if there is no currently running target_position
+        if hasattr(self, "_flm_move_future") and self._flm_move_future._state == RUNNING:
+            logging.debug("The stage is still moving, this movement isn't performed.")
+            return
+
+        if hasattr(self, "_flm_move_future") and not self._flm_move_future.button.Enabled:
+            logging.error("The stream is still updating and hence button activity is suppressed.")
+            return
+
+        evt.theButton.Enabled = False
+
+        @call_in_wx_main
+        def move_done(future):
+            self._tilted_3_DOF.updateMetadata({model.MD_FAV_POS_ACTIVE: self._tilted_3_DOF.position.value})
+            updated_stage_pos = self._stage.getMetadata()[model.MD_FAV_POS_ACTIVE]
+            updated_stage_pos.update({"x": self._stage.position.value["x"], "y": self._stage.position.value["y"]})
+            self._stage.updateMetadata({model.MD_FAV_POS_ACTIVE: updated_stage_pos})
+            future.button.Enabled = True
+
+        self._flm_move_future = self._tilted_3_DOF.moveRel({axis: proportion * self.tab_data_model.step_size.value})
+        self._flm_move_future.button = evt.theButton
+        self._flm_move_future.add_done_callback(move_done)
+
+    @call_in_wx_main
+    def _set_top_and_bottom_stream_and_settings(self, top, bottom):
+        """
+        Sets the top and bottom view and stream bar in a 2*1 viewport.
+
+        :param top (dict): The top stream and corresponding viewport accessible via the keys 'stream'/'viewport'
+        :param bottom(dict): The bottom stream and corresponding viewport accessible via the keys 'stream'/'viewport'
+        """
+        if top is bottom:
+            raise ValueError("The bottom stream is equal to the top stream, this isn't allowed in a 2*1 mode.")
+
+        # Pause all streams
+        self._opt_stream.is_active.value = False
+        self._sem_stream.is_active.value = False
+        self._fib_stream.is_active.value = False
+
+        for view in self.panel.pnl_two_streams_grid.viewports:
+            if view is top['viewport'] or view is bottom['viewport']:
+                view.Shown = True
+            else:
+                view.Shown = False
+        self.view_controller._grid_panel.set_shown_viewports(top['viewport'], bottom['viewport'])
+        self.tab_data_model.visible_views.value = [top['viewport'].view, bottom['viewport'].view]
+
+        # Alter the order of the view_ports so that the correct stream is displayed at top/bottom
+        viewport_list = self.panel.pnl_two_streams_grid.viewports
+        viewport_list.remove(top['viewport'])
+        viewport_list.insert(0, top['viewport'])
+        viewport_list.remove(bottom['viewport'])
+        viewport_list.insert(1, bottom['viewport'])
```

Comment:
This could be rewritten as `viewport_list = [top['viewport'], bottom['viewport'], viewport_list[2]]`